crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -549,31 +433,25 @@ impl<T: TransportConnect> Scheduler<T> {
 
 /// Placement hints for the [`logs_controller::LogsController`] based on the current
 /// [`SchedulingPlan`].
-pub struct SchedulingPlanNodeSetSelectorHints<'a> {
-    scheduling_plan: Option<&'a SchedulingPlan>,
+pub struct PartitionTableNodeSetSelectorHints {
+    partition_table: Pinned<PartitionTable>,
 }
```

Comment:
Is this struct kept across await points? If yes, then keeping a `Pinned` value can be a bit problematic because other users might run out of slots. So if one keeps a reference for a bit longer, then taking a snapshot is recommended.

Reply:
Thank you for pointing this one out. Right now it's not held across await points. It's either created and called immediately or passed to a non async function. So I think it's safe to keep this implementation which should be a little bit performant than a snapshot.